include/nbl/system/ISystem.h

```diff
@@ -16,14 +16,14 @@ namespace system
 class ISystem final : public core::IReferenceCounted
 {
 public:
+    static core::smart_refctd_ptr<ISystem> create();
```

Comment:
nope, you dont want the creator in the interface, the user will still need to know what implementation to construct.